Mathlib/Data/Complex/Basic.lean

```diff
@@ -81,39 +82,53 @@ theorem range_im : range im = univ :=
   im_surjective.range_eq
 #align complex.range_im Complex.range_im
 
+-- Porting note: refactored instance to allow `norm_cast` to work
+@[coe]
+def ofReal' (r : ℝ) : ℂ :=
+  ⟨r, 0⟩
 instance : Coe ℝ ℂ :=
-  ⟨fun r => ⟨r, 0⟩⟩
+  ⟨ofReal'⟩
 
-@[simp, norm_cast]
-theorem of_real_re (r : ℝ) : (r : ℂ).re = r :=
+/- Porting note: `simp` attribute removed as this has a variable as head symbol of
+the left-hand side (after whnfR)-/
+@[norm_cast]
+theorem ofReal_re (r : ℝ) : Complex.re (r : ℂ) = r :=
   rfl
-#align complex.of_real_re Complex.of_real_re
+#align complex.of_real_re Complex.ofReal_re
 
 @[simp, norm_cast]
-theorem of_real_im (r : ℝ) : (r : ℂ).im = 0 :=
+theorem ofReal_im (r : ℝ) : (r : ℂ).im = 0 :=
   rfl
-#align complex.of_real_im Complex.of_real_im
+#align complex.of_real_im Complex.ofReal_im
 
-theorem of_real_def (r : ℝ) : (r : ℂ) = ⟨r, 0⟩ :=
+-- Was warned that this is a syntactic tautology.
+@[nolint synTaut]
+theorem ofReal_def (r : ℝ) : (r : ℂ) = ⟨r, 0⟩ :=
   rfl
-#align complex.of_real_def Complex.of_real_def
+#align complex.of_real_def Complex.ofReal_def
 
+-- @[simp]
+/- Porting note: `simp` attribute removed as the result could be proved
+by `simp only [Complex.mk.injEq, and_true]` -/
```

Comment:
This is out of date, since you didn't actually remove the `simp` attribute.

Reply:
Thanks